src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -323,7 +319,7 @@ private static Binder AdjustBinderForPositionWithinStatement(int position, Binde
                     break;
 
                 case SyntaxKind.ForEachStatement:
-                case SyntaxKind.ForEachComponentStatement:
+                case SyntaxKind.ForEachVariableStatement:
                     var foreachStmt = (CommonForEachStatementSyntax)stmt;
                     if (LookupPosition.IsBetweenTokens(position, foreachStmt.OpenParenToken, foreachStmt.Statement.GetFirstToken()))
```

Comment:
>OpenParenToken [](start = 77, length = 14)

Perhaps we wouldn't need a special case above if we were to use in-keyword here instead.